openapi-cli/src/main/java/io/ballerina/openapi/generators/client/mime/MultipartFormData.java

```diff
@@ -0,0 +1,177 @@
+        this.imports = imports;
+        this.ballerinaUtilGenerator = ballerinaUtilGenerator;
+    }
+
+    @Override
+    public void setPayload(List<StatementNode> statementsList, Map.Entry<String, MediaType> mediaTypeEntry) {
+        ballerinaUtilGenerator.setRequestBodyMultipartFormDatafound(true);
+        ImportDeclarationNode mimeImport = GeneratorUtils.getImportDeclarationNode(BALLERINA, MIME);
+        if (!checkImportDuplicate(imports, MIME)) {
+            imports.add(mimeImport);
+        }
+        VariableDeclarationNode encodingMap = getMultipartMap(mediaTypeEntry);
+
+        VariableDeclarationNode bodyPartsVariable;
+        if (encodingMap == null) {
+            bodyPartsVariable = GeneratorUtils.getSimpleStatement("mime:Entity[]", "bodyParts",
+                    "check createBodyParts(payload)");
+        } else {
+            statementsList.add(encodingMap);
+            bodyPartsVariable = GeneratorUtils.getSimpleStatement("mime:Entity[]", "bodyParts",
+                    "check createBodyParts(payload, encodingMap)");
+        }
+
+        statementsList.add(bodyPartsVariable);
+        ExpressionStatementNode setPayloadExpression = GeneratorUtils.getSimpleExpressionStatementNode(
+                "request.setBodyParts(bodyParts)");
+        statementsList.add(setPayloadExpression);
+    }
+
+    public VariableDeclarationNode getMultipartMap(Map.Entry<String, MediaType> mediaTypeEntry) {
+        List<Node> filedOfMap = new ArrayList<>();
+        BuiltinSimpleNameReferenceNode mapType = createBuiltinSimpleNameReferenceNode(null,
+                createIdentifierToken("map<Encoding>"));
+        CaptureBindingPatternNode bindingPattern = createCaptureBindingPatternNode(
+                createIdentifierToken("encodingMap"));
+        TypedBindingPatternNode bindingPatternNode = createTypedBindingPatternNode(mapType, bindingPattern);
+
+        if (mediaTypeEntry.getValue().getEncoding() != null) {
+            Set<Map.Entry<String, Encoding>> endcodingEntries = mediaTypeEntry.getValue().getEncoding().entrySet();
+            for (Map.Entry<String, Encoding> entry : endcodingEntries) {
+                SpecificFieldNode contentField = null;
+                SpecificFieldNode headersField = null;
+                IdentifierToken fieldName = createIdentifierToken('"' + entry.getKey() + '"');
+                SeparatedNodeList<MappingFieldNode> separatedNodeList = createSeparatedNodeList();
+
+                if (entry.getValue().getContentType() != null) {
+                    contentField = createSpecificFieldNode(null,
+                            createIdentifierToken("contentType"), createToken(COLON_TOKEN),
+                            createRequiredExpressionNode(createIdentifierToken('"' +
+                                    entry.getValue().getContentType().split(",")[0] + '"')));
+                }
+
+                if (!entry.getValue().getHeaders().isEmpty()) {
```

Comment:
What about moving logic inside this `if clause` to a different function for readability?

Reply:
Added in https://github.com/ballerina-platform/openapi-tools/commit/b073cda18ba34b0a9e050f7cc820da3107ffa33b